### PR TITLE
Check path

### PR DESCRIPTION
**Pull request by tbarabosch**
This implements the check_path analyzer, which finds pathes from input functions (such as recv and scanf) to CWE hits.

### CODE REVIEW

File: src/analysis/check_path.ml [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+                              | _ -> false
+                              end)
+
+let collect_callsites program t =
+  Term.enum sub_t program
+  |> Seq.map ~f:(fun s -> if Term.enum blk_t s |>
+                               Seq.exists ~f:(fun b -> block_has_callsite b t) then Some s else None)
+  |> Seq.filter ~f:(fun s -> match s with
+                             | None -> false
+                             | _ -> true)
```
Comment (Enkelmann) [inline src/analysis/check_path.ml]: Isn't there a filter_map method for Seq.t for combining both?

File: src/utils/log_utils.ml [hunk 2]
```diff
@@ -5,57 +5,91 @@ module CweWarning = struct
...
   type t = {
       binary : string;
       time : float;
       warnings : CweWarning.t list;
+      check_path : CheckPath.t list;
     } [@@deriving yojson]
 end
 
 let cwe_warning_store = ref [||]
+let check_path_store = ref [||]
```
[the remark this answers is not in the document]
Comment (tbarabosch) [reply src/utils/log_utils.ml]: Done.

File: src/utils/log_utils.mli [hunk 3]
```diff
@@ -16,22 +16,37 @@
...
+      path : string list;
+      path_str : string;
+    }
 end
 
 val turn_off_logging : unit -> unit
 
-val cwe_warning_factory : string -> string -> ?other:string list list -> ?addresses:string list -> ?symbols:string list -> string -> CweWarning.t
+val cwe_warning_factory : string -> string -> ?other:string list list -> ?addresses:string list -> ?tids:string list -> ?symbols:string list -> string -> CweWarning.t
+val check_path_factory : ?path:string list -> ?path_str:string -> string -> string -> string -> string -> CheckPath.t
```
[the remark this answers is not in the document]
Comment (tbarabosch) [reply src/utils/log_utils.mli]: Done.

File: src/analysis/check_path.ml (hunk 1 above)
Comment (tbarabosch) [reply src/analysis/check_path.ml]: Done.

**Comment by tbarabosch** [comment]
Please ignore the nagging of codacy. These are just tests!

**Comment by tbarabosch** [comment]
> Very nice! If you approve of the documentation I added to _index.mld_ for the new command line option, we can merge it.

Looks fine.

**Pull request merged**